gnovm/cmd/gno/util.go

```diff
@@ -100,6 +100,64 @@ func gnoPackagesFromArgs(args []string) ([]string, error) {
 	return paths, nil
 }
 
+func gnoPackagePathsFromPattern(patterns []string) ([]string, error) {
+	paths := []string{}
+	recursiveLookup := false
+	for _, p := range patterns {
+		// TODO:
+		//     - Support more patterns
+		//     - Move this logic to separate helper function?
+		if strings.HasSuffix(p, "/...") {
+			recursiveLookup = true
+			p = strings.TrimSuffix(p, "/...")
+		}
+
+		info, err := os.Stat(p)
+		if err != nil {
+			return nil, fmt.Errorf("invalid file or package path: %w", err)
+		}
+
+		// if the passed arg is a file, then we'll just add it to the list
+		if !info.IsDir() {
+			paths = append(paths, p)
+			continue
+		}
+		// if the passed arg is just dir not followed by /...
+		if !recursiveLookup {
+			paths = append(paths, p)
+			continue
+		}
+
+		// the passed arg is a dir folllowed by /... then we'll walk the dir recursively
+		// and look for directories containing at least one .gno file.
+		visited := map[string]bool{} // used to run the builder only once per folder.
+		err = filepath.WalkDir(p, func(curpath string, f fs.DirEntry, err error) error {
+			if err != nil {
+				return fmt.Errorf("%s: walk dir: %w", p, err)
+			}
+			if f.IsDir() {
+				return nil // skip
+			}
+			if !isGnoFile(f) {
+				return nil // skip
+			}
+
+			parentDir := filepath.Dir(curpath)
+			if _, found := visited[parentDir]; found {
+				return nil
+			}
```

Comment:
These can probably be combined into a single check

Reply:
Done: 6a74eba625ed42b7c556bbab47299292312bec16